Add `Lightning service providers` page to `How it works` section (#568)
BitcoinDesign/Guide: guide/how-it-works/lightning-service-providers.md

> how-it-works/lightning-service-providers---

Reviewer: ```suggestion
---
```

Reviewer: This stray text breaks the page and makes the URI unreachable.

Lightning service providers (LSPs) solve these user friction points by providing services that make it easy to access and use the lightning network. They act as trust-minimized third parties that make it easy to onboard new user into lightning.

Roy Sheinfeld from [Breez](https://breez.technology/) explains LSPs in more depth in his [introducing lightning service providers](https://medium.com/breez-technology/introducing-lightning-service-providers-fe9fb1665d5f) and [envisioning LSPs in the lightning economy](https://medium.com/breez-technology/envisioning-lsps-in-the-lightning-economy-832b45871992) articles.

## What is a lightning service?

Lightning services are trust-minimized offerings by LSPs that solve user pain points when using a non-custodial bitcoin lightning wallet.

> A lightning service should always non-custodial, be opt-out, only used when necessary and any trade-offs should be clearly communicated to users.

Reviewer 2: This sentence is a bit hard to read. Was this the point of the sentence?
```suggestion
A lightning service should always be non-custodial, opt-out, with trade-offs clearly communicated to users.
```

Reviewer: ```suggestion
A lightning service should always be non-custodial and opt-out, with trade-offs clearly communicated to users.
```
I'd like to take the suggestion from @pavlenex a step further. "And" is better than a comma in this context.

Below are some lightning services, the pain points they solve, their trade-offs, and the applications that use them.

## Backup & recovery

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/how-it-works/lightning-service-providers/backups-recovery.jpg"
   retina = "/assets/images/guide/how-it-works/lightning-service-providers/backups-recovery@2x.jpg"
   alt-text = "Backups & recovery LSP service image."
   width = 400
   height = 300
   layout = "float-right-desktop"
%}

A Lightning wallet requires both a recovery phrase and up-to-date payment channel states to be recovered. Payment channel states need to be backed up each time a payment is received or sent. Users can make these backups, though if done incorrectly or stored insecurely, users are at risk of losing their bitcoin.

> A backup & recovery service automatically backs up and stores a users channel states with the LSP when a payment is sent or received. These backups are encrypted, usually by the user’s recovery phrase, before being sent to the LSP.

Reviewer 2: ```suggestion
A backup & recovery service automatically backs up and stores a user’s channel states with the LSP when a payment is sent or received. These backups are encrypted, usually by the user’s recovery phrase, before being sent to the LSP.
```

Author: I need to stop making this typo 😆

When a user recovers their wallet, they enter their recovery phrase, the LSP sends them the encrypted channel states, these are decrypted by the user, and the wallet is restored.

> This service requires trust that the LSP will make accurate, and securely store, the users channel state backups. Trusting your backups with a single third-party like this is also a security risk and makes users reliant on the LSP.

Reviewer 2: This sentence felt like it talked to the user instead of app maker, little update  to make sure we're speaking to the target audience of ours.
```suggestion
This service requires trust that the LSP accurately and securely stores the user's channel state backups. Entrusting a backup with a single third-party like this is also a security risk and makes users reliant on the LSP.
```

All of this can be complex for users to deal with. A channel opening service automatically opens a channel to the LSP for the user when necessary.

Users should always have the option to manually open channels, as to not completely rely on a LSP.

Below, we cover the different ways in which a channel open service can be offered, depending on user needs and the trade-offs they are willing to make.

</div>

### Inbound liquidity

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/how-it-works/lightning-service-providers/inbound-liquidity.jpg"
   retina = "/assets/images/guide/how-it-works/lightning-service-providers/inbound-liquidity@2x.jpg"
   alt-text = "Drop of water representing liquidity."
   width = 140
   height = 140
   layout = "float-right"
%}

Lightning wallets can not receive payments without inbound liquidity in a channel. This is primarily the case with new wallets, but may also occur when channels are imbalanced on funded wallets.

> Offering inbound liquidity with a channel open service addresses this situation. When a user receives a payment without enough inbound liquidity, a channel is automatically opened. The network fee, and sometimes an additional service fee, for opening the channel is typically deducted from the payment amount.

Reviewer 2: > inbound liquidity

This would be a great thing to add to glossary I think as we don't explain it anywhere in the guide. Just leaving it here for the reference.

Author: We have an issue up to create a whole page on this topic https://github.com/BitcoinDesign/Guide/issues/685,  but agree more lightning terminology needs to be defined in the glossary.

   alt-text = "Blocks representing unconfirmed spend."
   width = 140
   height = 140
   layout = "float-right"
%}

Using a channel open service requires a LSP to open a channel for the user on their behalf, which requires trust. An LSP could refuse to open a channel for a user or require invasive privacy practices, leading to application-level censorship.

A [dual-funding](https://bitcoinops.org/en/topics/dual-funding/) channel open service solves this issue of trust by having users cooperatively open a channel with the LSP. Users get the benefits of having both inbound and outbound liquidity without having to trust the LSP.

Dual funding is only applicable when funding a wallet with on-chain bitcoin. The issue of trusting the LSP still exists if funding with a lightning payment.

</div>

### Spend-unconfirmed

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/how-it-works/lightning-service-providers/spend-unconfirmed.jpg"
   retina = "/assets/images/guide/how-it-works/lightning-service-providers/spend-unconfirmed@2x.jpg"
   alt-text = "Blocks representing unconfirmed spend."
   width = 140
   height = 140
   layout = "float-right"
%}

> The bitcoin in a payment channel can not be spent until the on-chain transaction that opened the channel is confirmed. If the Mempool is congested, this could take a long time.

Reviewer 2: Good place to link to glossary. Also I don't think there's the mempool, only a mempool, right?
```suggestion
The bitcoin in a payment channel can not be spent until the on-chain transaction that opened the channel is confirmed. If a [mempool]({{ '/guide/nboarding/backing-up-a-wallet/cloud-backup/' | relative_url }}) is congested, this could take a long time.
```

Reviewer: I always hear people refer to it as "the mempool", even though technically each node has its own mempool. Regardless, lowercase is probably in order.

Reviewer: The glossary refers to it as "the mempool". Let's be consistent.

Reviewer: Also, the URL in the suggestion doesn't go to the glossary.

Reviewer: ```suggestion
The bitcoin in a payment channel can not be spent until the on-chain transaction that opened the channel is confirmed. If the [mempool]({{ '/guide/glossary/#mempool' | relative_url }}) is congested, this could take a long time.
```

Author: To avoid complexity, we could just say if the network is congested

A channel open service using spend-unconfirmed, sometimes called turbo channels or zero-confirmation channels, allows users to spend their bitcoin instantly without having to wait for the channel transaction to confirm.

> A spend-unconfirmed channel is trusted until the transaction is confirmed. Users should have the option to disable these type of channel opens if they are being used. Some LSPs may batch their channel opens, meaning a users channel will be trusted until that batch process is complete.

Reviewer 2: ```suggestion
A spend-unconfirmed channel is trusted until the transaction is confirmed. Users should have the option to disable these types of channel opens if they are being used. Some LSPs may batch their channel opens, meaning a user's channel will be trusted until that batch process is complete.
```
